be/src/exec/vectorized/hdfs_scanner_text.cpp

```diff
@@ -32,6 +42,12 @@ Status HdfsTextScanner::HdfsScannerCSVReader::_fill_buffer() {
         // is valid, according the RFC, add the record delimiter ourself.
         _buff.append(_record_delimiter);
     }
+
+    if ((_remain_length < 0 && _buff.find(_record_delimiter, 0) != nullptr)
+        || (_offset >= _file_length)) {
```

Comment:
1. would you mind add some comment for this logic?

If only offset 0 is supported, the remain_length will be the same with file_length? Otherwise some data won't be read

Reply:
> 
> If only offset 0 is supported, the remain_length will be the same with file_length? Otherwise some data won't be read

assume that we have only one  scan range:
123,123,xxx
this record will read by logic before here. should_stop_scan will be set, then in the next round the reader will return eof status. so i think no data will be missed.